tests/plugins/trim_whitespace.test.ts

```diff
@@ -0,0 +1,97 @@
+import { Model } from "../../src";
+import { InformationNotification } from "../../src/types";
+import { getCellContent } from "../test_helpers";
+import { setCellContent, setSelection } from "../test_helpers/commands_helpers";
+import { getRangeValuesAsMatrix, mountSpreadsheet } from "../test_helpers/helpers";
+
+let model: Model;
+
+// prettier-ignore
+
+describe("trim whitespace", () => {
+  test("trim cell content", () => {
+    model = new Model();
+    setCellContent(model, "A2", "   Alo         ");
+    setSelection(model, ["A2"]);
+    model.dispatch("TRIM_WHITESPACE");
+    expect(getCellContent(model, "A2")).toBe("Alo");
+  });
+
+  test("remove duplicate spaces", () => {
+    model = new Model();
+    setCellContent(model, "A2", "  Alo        salut     sunt  eu    un haiduc  ");
+    setSelection(model, ["A2"]);
+    model.dispatch("TRIM_WHITESPACE");
+    expect(getCellContent(model, "A2")).toBe("Alo salut sunt eu un haiduc");
+  });
+
+  test("keep lines break", () => {
+    model = new Model();
+    setCellContent(model, "A2", "  Alo        salut   \n   sunt  eu  \n  un haiduc  ");
+    setSelection(model, ["A2"]);
+    model.dispatch("TRIM_WHITESPACE");
+    expect(getCellContent(model, "A2")).toBe("Alo salut\nsunt eu\nun haiduc");
+  });
+
+  test("apply it on all selected cells", () => {
+    const cells = {
+        A2: { content: " a " }, 
+        A3: { content: " b " }, 
+        A4: { content: " c " }, 
+      };
+    model = new Model({ sheets: [{ cells }] });
+    setSelection(model, ["A2:A3", "A3:A4"]);
+    model.dispatch("TRIM_WHITESPACE");
+    expect(getRangeValuesAsMatrix(model, "A2:A4")).toEqual([["a"], ["b"], ["c"]]);
+  });
+});
+
+describe("notify user", () => {
+  test("notify when cells trimmed", async () => {
+    let notifyUserTextSpy = jest.fn();
+    const notifyUser = (notification: InformationNotification) => {
+      notifyUserTextSpy(notification.text);
+    };
+
+    const cells = {
+      A1: { content: " A B     B    A  " },
+      A2: { content: "  SPACES   INVADERS     " },
+      A3: { content: "NO SPACES TO TRIM" },
+    };
+    const { model } = await mountSpreadsheet(
+      {
+        model: new Model({ sheets: [{ cells }] }),
+      },
+      { notifyUser }
+    );
```

Comment:
You can do that to avoid having to mount a whole srpeadsheet:
```js
    const plugin = getPlugin(model, DataCleanupPlugin);
    jest.spyOn(plugin["ui"], "notifyUI").mockImplementation(notifyUserTextSpy);
```
Not the cleanset but I don't think we can mock an Object's property with our version of jest

Reply:
thx @hokolomopo :)

Reply:
Even shorter: 

```js
jest.spyOn(model.config, "notifyUI").mockImplementation(notifyUserTextSpy);
```